my-sms/spec/factories/institutes.rb

```diff
@@ -6,9 +6,10 @@
   factory :institute do
     transient { with_courses { false } }
     sequence(:name) { |n| "Institute #{n}" }
-  end
-  trait :demo do
-    name { [Faker::Educator.university, Faker::Educator.secondary_school].sample }
-    courses { with_courses ? Array.new(rand(8)) { association(:course) } : [] }
+    courses { with_courses ? Array.new(rand(8)) { association(:course, :no_institute) } : [] }
+    trait :demo do
+      name { [Faker::Educator.university, Faker::Educator.secondary_school].sample }
+      courses { with_courses ? Array.new(rand(8)) { association(:course, :demo, :no_institute) } : [] }
```

Comment:
Another tip: maybe just have a transient var for `courses_count`, and then you can just do

```ruby
courses { Array.new(courses_count) { assocation(:course, :demo, institutes: [instance]) }
```

taken from the HABTM guide on factory bot: https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#has_and_belongs_to_many-associations

Reply:
AH, just realised it's not a HABTM., in that case, you can omit the `institutes: [instance]` from above

Reply:
I saw that, just wanted to make it a random number of courses for more natural behavior. I thought about putting the `courses_count` inside `rand(n)` but it didn't really seem useful.